src/lib.rs

```diff
@@ -89,6 +89,59 @@ impl<K: std::hash::Hash + PartialEq, V> HashTable<K, V> {
     }
 }
 
+pub struct HashTableIterator<'a, K, V> {
+    current_bucket: Option<&'a Vec<(K, V)>>,
+    elements_iterator: Box<dyn Iterator<Item = &'a (K, V)> + 'a>,
+    buckets_iterator: Box<dyn Iterator<Item = &'a Vec<(K, V)>> + 'a>,
+}
+
+impl<'a, K, V> IntoIterator for &'a HashTable<K, V> {
+    type Item = &'a (K, V);
+
+    type IntoIter = HashTableIterator<'a, K, V>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        let mut buckets_iterator = self.buckets.iter();
+        let current_bucket = buckets_iterator.next();
+        // first elements iterator needs to be initialized
+        let elements_iterator = match current_bucket {
+            Some(b) => b.iter(),
+            None => [].iter(),
```

Comment:
I find this odd. Is there another way to return an empty iterator without needing to crate an empty slice?